### PR TITLE
resolved: minor tweaks to transaction logic and logging

### PR DESCRIPTION
**Pull request by poettering**
Nothing that matters. Post-v248 material I guess.

### CODE REVIEW

**Comment by yuwata** [comment]
LGTM.

File: src/resolve/resolved-dns-transaction.c [hunk 1]
```diff
@@ -1941,23 +1941,20 @@ int dns_transaction_go(DnsTransaction *t) {
...
                 case DNS_PROTOCOL_LLMNR:
-                        jitter %= LLMNR_JITTER_INTERVAL_USEC;
+                        jitter = random_u64_range(LLMNR_JITTER_INTERVAL_USEC);
                         accuracy = LLMNR_JITTER_INTERVAL_USEC;
                         break;
 
                 case DNS_PROTOCOL_MDNS:
-                        jitter %= MDNS_JITTER_RANGE_USEC;
-                        jitter += MDNS_JITTER_MIN_USEC;
+                        jitter = usec_add(random_u64_range(MDNS_JITTER_RANGE_USEC), MDNS_JITTER_MIN_USEC);
```
Comment (keszybz) [inline src/resolve/resolved-dns-transaction.c:1958]: I now submitted a test for random_u64_range() in another PR, and it seems to work correctly ;)

File: src/resolve/resolved-dns-transaction.c [hunk 2]
```diff
@@ -1962,11 +1962,13 @@ int dns_transaction_go(DnsTransaction *t) {
...
 
-                r = sd_event_add_time(
+                assert(!t->timeout_event_source);
+
+                r = sd_event_add_time_relative(
                                 t->scope->manager->event,
                                 &t->timeout_event_source,
                                 clock_boottime_or_monotonic(),
-                                ts + jitter, accuracy,
+                                jitter, accuracy,
```
Comment (keszybz) [inline src/resolve/resolved-dns-transaction.c:1971]: This moves the next base of next attempt from sd_event_now(), i.e. from the most recent event loop iteration beginning, to "now". I guess that's appropriate.

**Review by keszybz — COMMENTED** [review]
LGTM too.

File: src/resolve/resolved-dns-transaction.c (hunk 2 above)
Comment (poettering) [reply src/resolve/resolved-dns-transaction.c:1971]: actually `ts` was initialized to sd_event_now() too, so this isn't changed. Besides some more careful overflow checking this code shouldn't be altered at all i think by this commit

**Pull request merged**